code/modules/mob/living/carbon/carbon.dm

```diff
@@ -686,4 +686,37 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 				handcuffed,
 				legcuffed,
 				back,
-				wear_mask)
\ No newline at end of file
+				wear_mask)
+
+/mob/living/carbon/proc/update_addictions(var/R, var/amount = 0)
+	if(isbrain(src) || isslime(src))		//Only human and xeno types can handle addictions
```

Comment:
This proc should probably be in the life.dm file, and I'm not too sure about having this check hardcoded here. Wouldn't a variable on the mob make more sense?